xcresult/src/xcresult.rs

```diff
@@ -0,0 +1,348 @@
+                });
+                if let Some(failure) = failure {
+                    let failure_message = failure
+                        .get("message")
+                        .and_then(|r| r.get("_value"))
+                        .and_then(|r| r.as_str());
+                    let failure_uri = failure
+                        .get("documentLocationInCreatingWorkspace")
+                        .and_then(|r| r.get("url"))
+                        .and_then(|r| r.get("_value"))
+                        .and_then(|r| r.as_str());
+                    testcase_status.set_message(failure_message.unwrap_or(""));
+                    uri = failure_uri.unwrap_or("").replace("file://", "");
+                }
+            }
+        }
+        let mut testcase_junit = TestCase::new(name, testcase_status);
+        let id = testcase
+            .get("identifierURL")
+            .and_then(|r| r.get("_value"))
+            .and_then(|r| r.as_str())
+            .map(XCResult::generate_id)
+            .unwrap_or_default();
+        testcase_junit.extra.insert("id".into(), id.into());
+        let file_components = uri.split('#').collect::<Vec<&str>>();
+        if file_components.len() == 2 {
+            testcase_junit
+                .extra
+                .insert("file".into(), file_components[0].into());
+        }
+        if let Some(classname) = testcase_group
+            .get("name")
+            .and_then(|r| r.get("_value"))
+            .and_then(|r| r.as_str())
+        {
+            testcase_junit.set_classname(classname);
+        }
+
+        if let Some(duration) = testcase
+            .get("duration")
+            .and_then(|r| r.get("_value"))
+            .and_then(|r| r.as_str())
+            .and_then(|r| r.parse::<f32>().ok())
+        {
+            testcase_junit.set_time(std::time::Duration::from_secs_f32(duration));
+        }
+        Ok(testcase_junit)
+    }
+
+    fn junit_testsuite(
+        &self,
+        action: &serde_json::Value,
+        testsuite: &serde_json::Value,
+    ) -> Result<TestSuite, anyhow::Error> {
+        let testsuite_name = testsuite
+            .get("name")
+            .and_then(|r| r.get("_value"))
+            .and_then(|r| r.as_str());
+        if testsuite_name.is_none() {
+            log::debug!("failed to get name of testsuite: {:?}", testsuite);
+            return Err(anyhow::anyhow!("failed to get name of testsuite"));
+        }
+        let mut testsuite_junit = TestSuite::new(testsuite_name.unwrap_or(""));
+        let mut index_map = indexmap! {};
+        if let Some(identifier) = testsuite
+            .get("identifierURL")
+            .and_then(|r| r.get("_value"))
+            .and_then(|r| r.as_str())
+        {
+            index_map.append(&mut indexmap! {
+                XmlString::new("id") => XmlString::new(XCResult::generate_id(identifier)),
+            });
+        }
+        testsuite_junit.extra = index_map;
```

Comment:
Similar [comment to before](https://github.com/trunk-io/analytics-cli/pull/106#discussion_r1783343394): no need to create a new `IndexMap` when we can just mutate `.extra` directly